test/sslapitest.c

```diff
@@ -1970,7 +1970,7 @@ static int alpn_select_cb (SSL *ssl, const unsigned char **out, unsigned char *o
     unsigned int i, protlen = 0;
     const unsigned char *prot;
 
-    for (i = 0, prot = in; i < inlen; i += protlen, prot += protlen) {
+    for (i = 0, prot = in; i < inlen - 1; i += protlen, prot += protlen) {
         protlen = *(prot++);
```

Comment:
This fix doesn't look right to me. Shouldn't there be an `i++` at the same time as we do the `prot++` here?

Reply:
sounds right, fixed...